src/crawler.ts

```diff
@@ -51,20 +56,51 @@ export async function handleCrawlerRequests(
   requests: GetCharacterRequestParams[]
 ) {
   const handleRequest = async function (req: GetCharacterRequestParams) {
-    const crawler = new WarmaneCrawler();
-    logger.info(`crawling started for ${req.name} on ${req.realm}`);
-    const matchDetails = await crawler.fetchAllMatchDetails({
-      character: req.name,
-      realm: req.realm,
-    });
-    const databaseWrites = matchDetails.map((match) =>
-      matchDetailsStore.upsert(match)
-    );
-    logger.info(`crawling completed for ${req.name} on ${req.realm}`);
-    await Promise.all(databaseWrites);
-    logger.info(
-      `crawler results saved successfully for ${req.name} on ${req.realm}`
-    );
+    const { name, realm } = req;
+    const characterId: CharacterId = `${name}@${realm}`;
+
+    try {
+      const crawler = new WarmaneCrawler();
+
+      await crawlerStateStore.upsert({
+        id: characterId,
+        state: "running",
+        crawler_last_started: new Date().toISOString(),
+      });
+      logger.info(`crawling started for ${req.name} on ${req.realm}`);
+
+      const matchDetails = await crawler.fetchAllMatchDetails({
+        character: req.name,
+        realm: req.realm,
+      });
+
+      const databaseWrites = matchDetails.map((match) =>
+        matchDetailsStore.upsert(match)
+      );
+
+      logger.info(`crawling completed for ${req.name} on ${req.realm}`);
+      await Promise.all(databaseWrites);
+      logger.info(
+        `crawler results saved successfully for ${req.name} on ${req.realm}`
+      );
+
+      await crawlerStateStore.upsert({
+        id: characterId,
+        state: "idle",
+        crawler_last_finished: new Date().toISOString(),
+      });
+    } catch (error) {
+      const stateUpdate: CrawlerState = {
+        id: characterId,
+        state: "errored",
+      };
+
+      if (error instanceof Error) {
+        stateUpdate.crawler_errors = [error.message];
```

Comment:
It doesn't seem to me that there's anything in place to automatically delete error messages. Though it's unlikely they could grow to a state that would be noticeable, let alone an issue, would it be a good idea to implement some form of management to delete error messages after a certain time period, or at a certain limit?

Reply:
I think it overwrites what was previously there.

`let errors = ["something failed"]`
`errors = ["something else failed"]`
`//["something else failed"]`

Having this typed as a string[] is more of a forward thinking design. Right now there is only one try/catch around the entire thing i.e. there is only one error message. It also means we have an all or nothing attempt to crawl all of the matches played. If someone has played 10,000 games and we fail to get just one, the entire operation is a waste.

I would like to see us be able to support a partial success, and perhaps we log out in an array all of the errors that may have occurred.